pkg/controller/common/annotation/timeout.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package annotation
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ExtractTimeout extracts a timeout value specified as an annotation on a resource.
+func ExtractTimeout(obj runtime.Object, annotation string, defaultVal time.Duration) time.Duration {
+	if obj == nil {
+		return defaultVal
+	}
+
+	metaAcc := meta.NewAccessor()
+
+	ann, err := metaAcc.Annotations(obj)
+	if err != nil {
+		log.V(1).Info("Failed to extract annotations from object", "error", err, "object", obj)
+		return defaultVal
+	}
```

Comment:
we could avoid all the meta accessor stuff if the function was passed a `metav1.ObjectMeta` instead of a `runtime.Object`?

Reply:
Good point.